### PR TITLE
Adjust swift-specific changes to match 67d67ebe8f2 on swift/master-next

### PR DESCRIPTION
**Pull request by bulbazord**
Looks like this was overlooked during the merge.

cc @JDevlieghere @jimingham @dcci

### CODE REVIEW

**Comment by jimingham** [comment]
Thanks for doing this.  


I was in the middle of doing this, but stopped to run the test suite after this merge, which failed altogether because it looks like we are always using the static bindings for Python and they are missing SBPlatform::GetHostPlatform which is used by the test suite.  As soon as I figure out how to make it NOT use the static bindings I’ll fix them…

Jim

**Review by dcci — COMMENTED** [review]
@jimingham I guess you're the only one who can review this change, so I'll leave it to you :)

File: lldb/source/Plugins/ExpressionParser/Swift/SwiftPersistentExpressionState.cpp
Comment (jimingham) [inline lldb/source/Plugins/ExpressionParser/Swift/SwiftPersistentExpressionState.cpp]: This isn't right.  You have to switch on is_error and either use the m_next_persistent_variable_id or the m_next_persistent_error_id.  Something like:

```+ConstString 
+SwiftPersistentExpressionState::GetNextPersistentVariableName(bool is_error)
+{
+  llvm::SmallString<64> name;
+  {
+    llvm::raw_svector_ostream os(name);
+    uint32_t variable_num = is_error ? m_next_persistent_error_id++ 
+                                    : m_next_persistent_variable_id++;
+    os << GetPersistentVariablePrefix(is_error)
+       << variable_num;
+  }
+  return ConstString(name);
+}
+
```

**Review by jimingham — CHANGES_REQUESTED** [review]
You need to fix GetNextPersistentVariableName to use the m_next_persistent_error_id, other than that this looks good.

File: lldb/source/Plugins/ExpressionParser/Swift/SwiftPersistentExpressionState.cpp
Comment (bulbazord) [reply lldb/source/Plugins/ExpressionParser/Swift/SwiftPersistentExpressionState.cpp]: Fixed

**Review by jimingham — APPROVED** [review]
LGTM

**Comment by bulbazord** [comment]
@jimingham Can you merge this for me? I don't have commit access to apple/llvm-project :)

**Comment by jimingham** [comment]
Done.

Jim

**Pull request merged**
